consensus/istanbul/backend/snapshot.go

```diff
@@ -220,22 +219,13 @@ func (s *Snapshot) apply(headers []*types.Header, gov governance.Engine, addr co
 			}
 		}
 	}
-	snap.Number += uint64(len(headers))
-	snap.Hash = headers[len(headers)-1].Hash()
-
-	if snap.ValSet.Policy() == istanbul.WeightedRandom {
-		snap.ValSet.SetBlockNum(snap.Number)
-
-		bigNum := new(big.Int).SetUint64(snap.Number)
-		if chain.Config().IsRandaoForkBlockParent(bigNum) {
-			// The ForkBlock must select proposers using MixHash but (ForkBlock - 1) has no MixHash. Using ZeroMixHash instead.
-			snap.ValSet.SetMixHash(params.ZeroMixHash)
-		} else if chain.Config().IsRandaoForkEnabled(bigNum) {
-			// Feed parent MixHash
-			snap.ValSet.SetMixHash(headers[len(headers)-1].MixHash)
-		}
-	}
```

Comment:
@blukat29 If you worry about this change, it can be applied after releasing v1.12.0